csharp/src/Ice/Communicator-LocatorManager.cs

```diff
@@ -76,17 +76,9 @@ public void Response(LocatorInfo locatorInfo, IObjectPrx? proxy)
 
             public void Exception(LocatorInfo locatorInfo, System.Exception exc)
             {
-                try
-                {
-                    locatorInfo.GetEndpointsException(_ref, exc); // This throws.
-                }
-                catch (System.Exception ex)
-                {
-                    if (_callback != null)
-                    {
-                        _callback.SetException(ex);
-                    }
-                }
+                // Call even when _callback is null to get the logging
+                System.Exception ex = GetEndpointsException(_ref, exc);
+                _callback?.SetException(ex);
```

Comment:
Maybe we can rename this as `void TraceEndpointsException` seems this always returns the same exception, or even move the trace code here as it is not used anywhere else

```suggestion
                TraceEndpointsException(_ref, ex);
                _callback?.SetException(ex);
```

Reply:
Fixed as suggested.